docs/guides/migration/pip-to-project.md

```diff
@@ -0,0 +1,420 @@
+tqdm==4.67.1
+    # via -r requirements.in
+```
+
+While on Windows, this compiles to:
+
+```text title="requirements-win.txt"
+colorama==0.4.6
+    # via tqdm
+tqdm==4.67.1
+    # via -r requirements.in
+```
+
+`colorama` is a Windows-only dependency of `tqdm`.
+
+When using `pip` and `pip-tools`, a project needs to declare a requirements lock file for each
+supported platform.
+
+!!! note
+
+        uv's resolver can compile dependencies for multiple platforms at once (see ["universal resolution"](../../concepts/resolution.md#universal-resolution)),
+        allowing you to use a single `requirements.txt` for all platforms:
+
+        ```console
+        $ uv pip compile --universal requirements.in
+        ```
+
+        ```text title="requirements.txt"
+        colorama==0.4.6 ; sys_platform == 'win32'
+            # via tqdm
+        tqdm==4.67.1
+            # via -r requirements.in
+        ```
+
+        This resolution mode is also used when using a `pyproject.toml` and `uv.lock`.
+
+## The `pyproject.toml`
+
+The `pyproject.toml` is a standardized file for Python project metadata. It replaces
+`requirements.in` files, allowing you to represent arbitrary groups of project dependencies. It also
+provides a centralized location for metadata about your project, such as the build system or tool
+settings.
+
+<!-- TODO: Link to the official docs on this or write more -->
+
+We can translate the example `requirements.in` and `requirements-dev.in` files to a `pyproject.toml`
+as follows:
+
+```toml title="pyproject.toml"
+[project]
+name = "example"
+version = "0.0.1"
+dependencies = [
+    "fastapi",
+    "pydantic>2"
+]
+
+[dependency-groups]
+dev = ["pytest"]
+```
+
+## `uv.lock`
+
+uv uses the `uv.lock` file to lock package versions. The format of this file is specific to uv,
+allowing uv to support advanced features. It replaces `requirements.txt` files. Unlike
+`requirements.txt` files, the `uv.lock` file can represent arbitrary groups of dependencies, so
+multiple files are not needed to lock development dependencies.
+
+To learn more, see the [lockfile](../../concepts/projects/layout.md#the-lockfile) documentation.
+
+## Importing requirements files
+
+The simpest way to import requirements is with `uv add`:
+
+```
+$ uv add -r requirements.in
+```
+
+However, there is some nuance to this transition. Notice we used the `requirements.in` file, which
+does not pin to exact versions of packages so uv will solve for new versions of these packages. You
+may want to continue using your previously locked versions from your `requirements.txt` so, when
+switching over to uv, none of your dependency versions change.
+
+The solution is to add your locked versions as _constraints_. uv supports using these on `add` to
+preserved locked versions:
+
+```
+$ uv add -r requirements.in -c requirements.txt
+```
+
+Your existing versions will be retained when producing a `uv.lock` file.
+
+### Importing platform-specific constraints
```

Comment:
Maybe it's a way less common use case, but would the same issue applies to Python versions too, for projects that need to support multiple versions of Python? And if so, do you think a specific section would be worth having?

Reply:
Yeah that problem would apply. I'm not sure 🤔. I don't know if I've seen many requirements files toggled on version. I can try to add a note of some sort.